### PR TITLE
Locks and queue for linking

### PR DESCRIPTION
**Pull request by geekbeast**
In Progress.

### CODE REVIEW

File: src/main/kotlin/com/openlattice/linking/controllers/RealtimeLinkingController.kt [hunk 1]
```diff
@@ -29,6 +25,9 @@ class RealtimeLinkingController(
     @Inject
     private lateinit var authz: AuthorizationManager
 
+    @Inject
+    private lateinit var backgroundLinkingService: BackgroundLinkingService
```
Comment (you-leee) [inline src/main/kotlin/com/openlattice/linking/controllers/RealtimeLinkingController.kt:29]: I don't think the RealtimeLinkingController needs the BackgroundLinkingService

File: src/main/kotlin/com/openlattice/linking/BackgroundLinkingService.kt [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+ * assigned to any real entity set.
+ */
+internal val LINKING_ENTITY_SET_ID = UUID(0, 0)
+internal const val PERSON_FQN = "general.person"
+internal const val REFRESH_PROPERTY_TYPES_INTERVAL_MILLIS = 30000L
+internal const val LINKING_BATCH_TIMEOUT = 120000L
+internal const val MINIMUM_SCORE = 0.75
+
+/**
+ * Performs realtime linking of individuals as they are integrated ino the system.
```
Comment (makosblade) [inline src/main/kotlin/com/openlattice/linking/BackgroundLinkingService.kt:60]: 🤔

**Review by makosblade — APPROVED** [review]
LGTM

File: src/main/kotlin/com/openlattice/linking/controllers/RealtimeLinkingController.kt (hunk 1 above)
Comment (geekbeast) [reply src/main/kotlin/com/openlattice/linking/controllers/RealtimeLinkingController.kt:29]: I am adding some APIs for controlling linking.

**Comment by geekbeast** [comment]
This ready to go, implementation of linking management apis will be in a future PR.

**Review by katherinebernstein — APPROVED** [review]

**Pull request merged**